src/algorithms/nsga2.cpp

```diff
@@ -201,34 +203,92 @@ population nsga2::evolve(population pop) const
 
         // 3 - We then loop thorugh all individuals with increment 4 to select two pairs of parents that will
         // each create 2 new offspring
-        for (decltype(NP) i = 0u; i < NP; i += 4) {
-            // We create two offsprings using the shuffled list 1
-            parent1_idx = tournament_selection(shuffle1[i], shuffle1[i + 1], ndr, pop_cd);
-            parent2_idx = tournament_selection(shuffle1[i + 2], shuffle1[i + 3], ndr, pop_cd);
-            crossover(child1, child2, parent1_idx, parent2_idx, pop);
-            mutate(child1, pop);
-            mutate(child2, pop);
-            // we use prob to evaluate the fitness so
-            // that its feval counter is correctly updated
-            auto f1 = prob.fitness(child1);
-            auto f2 = prob.fitness(child2);
-            popnew.push_back(child1, f1);
-            popnew.push_back(child2, f2);
-
-            // We repeat with the shuffled list 2
-            parent1_idx = tournament_selection(shuffle2[i], shuffle2[i + 1], ndr, pop_cd);
-            parent2_idx = tournament_selection(shuffle2[i + 2], shuffle2[i + 3], ndr, pop_cd);
-            crossover(child1, child2, parent1_idx, parent2_idx, pop);
-            mutate(child1, pop);
-            mutate(child2, pop);
-            // we use prob to evaluate the fitness so
-            // that its feval counter is correctly updated
-            f1 = prob.fitness(child1);
-            f2 = prob.fitness(child2);
-            popnew.push_back(child1, f1);
-            popnew.push_back(child2, f2);
-        } // popnew now contains 2NP individuals
-
+        if (m_bfe) {
+          // bfe is available:
+          auto n_obj = prob.get_nobj();
+          std::vector<vector_double> poptemp;
+          std::vector<unsigned long> fidtemp;
+          for (decltype(NP) i = 0u; i < NP; i += 4) {
+              // We create two offsprings using the shuffled list 1
+              parent1_idx = tournament_selection(shuffle1[i], shuffle1[i + 1], ndr, pop_cd);
+              parent2_idx = tournament_selection(shuffle1[i + 2], shuffle1[i + 3], ndr, pop_cd);
+              crossover(child1, child2, parent1_idx, parent2_idx, pop);
+              mutate(child1, pop);
+              mutate(child2, pop);
+
+              poptemp.push_back(child1);
+              poptemp.push_back(child2);
+              //popnew.push_back(child1, f1);
```

Comment:
@willcodeforfudd can these commented lines be removed?

Reply:
Yes, of course.   Sorry for the sloppy code, and repeated commits - this is my first time collaborating on a github project.

Reply:
@willcodeforfudd no worries, thanks for contributing the code. I am trying to ping @darioizzo to see if he can give the final OK for merging.